### PR TITLE
add e2e test

### PR DESCRIPTION
**Pull request by akoserwal**
## Describe your changes

- ...

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -68,7 +68,7 @@ clean:
 test:
 	@echo ""
 	@echo "Running tests."
-	@go test ./... -count=1 -coverprofile=coverage.out
+	@go test ./... -count=1 -coverprofile=coverage.out -skip 'TestInventoryAPIGRPC_*|Test_ACMKafkaConsumer'
```
Comment (jmelis) [inline Makefile:71]:
```suggestion
	# TODO: e2e tests are taking too long to be enabled by default. They need to be sped up.
	@go test ./... -count=1 -coverprofile=coverage.out -skip 'TestInventoryAPIGRPC_*|Test_ACMKafkaConsumer'
```

File: test/e2e-batch.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      containers:
+        - name: inventory-e2e-tests
+          image: localhost/inventory-e2e-tests:latest
+          env:
+            - name: INV_GRPC_URL
+              value: "inventory.multicluster-global-hub.svc.cluster.local:9081"
+            - name: KAFKA_BOOTSTRAP_SERVERS
+              value: "kafka-kafka-bootstrap.multicluster-global-hub:9092"
+      restartPolicy: Never
+  backoffLimit: 1
```
Comment (jmelis) [inline test/e2e-batch.yaml:17]: won't this need to mount the kafka creds?

**Review by jmelis — COMMENTED** [review]
it looks good, but I'm not sure how the test container will obtain the creds to authenticate with kafka and with inventory

File: test/e2e-batch.yaml (hunk 2 above)
Comment (akoserwal) [reply test/e2e-batch.yaml:17]: yes, I couldn't get the global hub set up through the script `e2e_setup_mcgh.sh`. I wasn't sure what configurations I need to configure for kafka.

**Comment by jmelis** [comment]
LGTM

**Comment by josejulio** [comment]
Rebased to latest main

**Pull request merged**